scripts/sync-discord-roles.ts

```diff
@@ -0,0 +1,262 @@
+import fs from "fs"
+import path from "path"
+import { Client, Role } from "discord.js"
+
+// ----- Role definitions ----- //
+
+// All-Time roles (by score)
+// These roles are cumulative: if a user qualifies for a higher tier, they should have
+// all lower-tier roles as well. However, WE DO NOT REMOVE ANY ALL-TIME ROLES ONCE ASSIGNED.
+const ALL_TIME_ROLE_NAMES = [
+  "New Contributor",
+  "Contributor",
+  "Try Hard",
+  "Hero",
+  "King",
+  "Legend",
+] as const
+type AllTimeRoleName = (typeof ALL_TIME_ROLE_NAMES)[number]
+
+// Weekly roles (by stars)
+// These roles are fully synced each week: only the cumulative weekly roles based on the current
+// stars value should be present.
+const WEEKLY_ROLE_NAMES = ["⭐", "⭐⭐", "⭐⭐⭐", "👑", "👑👑"] as const
+type WeeklyRoleName = (typeof WEEKLY_ROLE_NAMES)[number]
+
+// ----- Helper function for All-Time roles ----- //
+
+function getAllTimeRoleName(score: number): AllTimeRoleName | null {
+  if (score >= 100) return "Legend"
+  if (score >= 75) return "King"
+  if (score >= 50) return "Hero"
+  if (score >= 30) return "Try Hard"
+  if (score >= 10) return "Contributor"
+  if (score >= 3) return "New Contributor"
+  return null
+}
+
```

Comment:
Can you import this function? Its being copied everywhere- there should be one source of truth they might change over time!

Reply:
Oh nvm i guess this maps differently? Is there a way to use an existing fn to pull the exact score per level?